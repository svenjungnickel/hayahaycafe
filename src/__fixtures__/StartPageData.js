export default {
    slug: 'slug',
    title: 'title',
    subtitle: 'subtitle',
    headerImage: {
        publicURL: 'publicURL',
    },
    gallery: [
        {
            src: {
                publicURL: 'publicURL',
            },
            alt: 'gallery image 01',
            title: 'Gallery Image 01',
        },
        {
            src: {
                publicURL: 'publicURL',
            },
            alt: 'gallery image 02',
            title: 'Gallery Image 02',
        },
    ],
    meta: {
        title: 'meta title',
        description: 'meta description.',
        keywords: 'meta keywords',
    },
};
