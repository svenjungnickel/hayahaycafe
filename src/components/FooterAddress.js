import React from 'react';
import PropTypes from 'prop-types';
import { graphql, StaticQuery } from 'gatsby';

const AddressItem = ({ item }) => <>{item && <>{item}, </>}</>;

AddressItem.propTypes = {
    item: PropTypes.string.isRequired,
};

const FooterAddress = () => (
    <StaticQuery
        query={graphql`
            query {
                globalSettings: settingsYaml {
                    location {
                        address {
                            company
                            addressLine1
                            addressLine2
                            province
                            postalCode
                            country
                        }
                    }
                }
            }
        `}
        render={({ globalSettings }) => (
            <>
                {!!globalSettings && !!globalSettings.location && !!globalSettings.location.address && (
                    <p>
                        <AddressItem item={globalSettings.location.address.company} />
                        <AddressItem item={globalSettings.location.address.addressLine1} />
                        <AddressItem item={globalSettings.location.address.addressLine2} />
                        <AddressItem item={globalSettings.location.address.province} />
                        <AddressItem item={globalSettings.location.address.postalCode} />
                        {globalSettings.location.address.country && <>{globalSettings.location.address.country}</>}
                    </p>
                )}
            </>
        )}
    />
);

export default FooterAddress;
