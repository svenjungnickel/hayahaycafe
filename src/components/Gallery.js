import React, { useState } from 'react';
import PropTypes from 'prop-types';
import Carousel from 'nuka-carousel';
import FsLightbox from 'fslightbox-react';
import Image from './Image';
import GalleryStyle from '../styles/components/Gallery.module.scss';
import '../styles/libs/fslightbox-react.scss';

const PreviousButton = ({ previousSlide }) => (
    <button className={GalleryStyle.previousButton} aria-label="previous" onClick={previousSlide}>
        &lsaquo;
    </button>
);

PreviousButton.propTypes = {
    previousSlide: PropTypes.func.isRequired,
};

const NextButton = ({ nextSlide }) => (
    <button className={GalleryStyle.nextButton} aria-label="next" onClick={nextSlide}>
        &rsaquo;
    </button>
);

NextButton.propTypes = {
    previousSlide: PropTypes.func.isRequired,
};

const PagingDots = ({ slideCount, slidesToScroll, goToSlide, currentSlide }) => (
    <ul className={GalleryStyle.pagingDots__list}>
        {PagingDots.getIndexes(slideCount, slidesToScroll).map(index => {
            return (
                <li className={GalleryStyle.pagingDots__listItem} key={index}>
                    <button
                        className={GalleryStyle.pagingDots__button}
                        style={{ opacity: currentSlide === index ? 1 : 0.5 }}
                        onClick={goToSlide.bind(null, index)}
                    >
                        &bull;
                    </button>
                </li>
            );
        })}
    </ul>
);

PagingDots.getIndexes = (count, inc) => {
    const arr = [];
    for (let i = 0; i < count; i += inc) {
        arr.push(i);
    }
    return arr;
};

PagingDots.propTypes = {
    slideCount: PropTypes.number.isRequired,
    slidesToScroll: PropTypes.number.isRequired,
    goToSlide: PropTypes.func.isRequired,
    currentSlide: PropTypes.number.isRequired,
};

const Gallery = ({ images }) => {
    const [toggler, setToggler] = useState(false);
    const imageSources = images.map(image => {
        if (!!image.src && !!image.src.childImageSharp) {
            if (image.src.childImageSharp.fluid) {
                return image.src.childImageSharp.fluid.src;
            }

            return image.src.childImageSharp.fixed.src;
        }

        return image.src;
    });

    return (
        <>
            <Carousel
                dragging={true}
                swiping={true}
                wrapAround={true}
                heightMode="first"
                renderCenterLeftControls={props => <PreviousButton {...props} />}
                renderCenterRightControls={props => <NextButton {...props} />}
                renderBottomCenterControls={props => <PagingDots {...props} />}
            >
                {images.map((image, index) => (
                    <Image
                        key={index}
                        src={image.src}
                        alt={image.alt}
                        title={image.title}
                        onClick={() => setToggler(!toggler)}
                    />
                ))}
            </Carousel>
            <FsLightbox toggler={toggler} sources={imageSources} />
        </>
    );
};

Gallery.propTypes = {
    images: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default Gallery;
