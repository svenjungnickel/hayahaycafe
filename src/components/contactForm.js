import React, { useState, createRef } from 'react';
import { Alert, Button, Col, Form } from 'react-bootstrap';
import Recaptcha from 'react-google-recaptcha';
import Fade from 'react-reveal/Fade';
import Styles from '../styles/components/contactForm.module.scss';

const RECAPTCHA_KEY = process.env.SITE_RECAPTCHA_KEY;
const CONTACT_FORM_FADE_UP_DURATION = 400;
const SUCCESS_MESSAGE_FADE_IN_DURATION = 200;

function encode(data) {
    return Object.keys(data)
        .map(key => encodeURIComponent(key) + '=' + encodeURIComponent(data[key]))
        .join('&');
}

export default () => {
    const [validated, setValidated] = useState(false);
    const [state, setState] = useState({});
    const [recaptchaError, setRecaptchaError] = useState(false);
    const [success, setSuccess] = useState(false);
    const recaptchaRef = createRef();

    const handleChange = e => {
        setState({ ...state, [e.target.name]: e.target.value });
    };

    const handleRecaptcha = () => {
        setRecaptchaError(false);
    };

    const handleSubmit = event => {
        event.preventDefault();
        event.stopPropagation();
        setValidated(true);

        const recaptchaValue = recaptchaRef.current.getValue() || false;
        if (false === recaptchaValue) {
            setRecaptchaError(true);
            return;
        }

        if (event.currentTarget.checkValidity() === false) {
            return;
        }

        const form = event.target;
        fetch('/', {
            method: 'POST',
            headers: { 'Content-Type': 'application/x-www-form-urlencoded' },
            body: encode({
                'form-name': form.getAttribute('name'),
                'g-recaptcha-response': recaptchaValue,
                ...state,
            }),
        })
            .then(() => {
                console.log('success');
                setSuccess(true);
            })
            .catch(error => alert(error));
    };

    return (
        <div className={Styles.contactForm}>
            <Fade top collapse when={false === success} duration={CONTACT_FORM_FADE_UP_DURATION} unmountOnExit={true}>
                <Form
                    noValidate
                    validated={validated}
                    onSubmit={handleSubmit}
                    name="contact"
                    method="POST"
                    action="/thanks"
                    data-netlify="true"
                    data-netlify-recaptcha="true"
                >
                    <Form.Label>Name *</Form.Label>
                    <Form.Row>
                        <Col xs={12} lg={6}>
                            <Form.Group controlId="contactFirstName">
                                <Form.Control name="firstName" type="text" required onChange={handleChange} />
                                <Form.Text className="text-muted">First name</Form.Text>
                                <Form.Control.Feedback type="invalid">
                                    Please enter your first name.
                                </Form.Control.Feedback>
                            </Form.Group>
                        </Col>
                        <Col xs={12} lg={6}>
                            <Form.Group controlId="contactLastName">
                                <Form.Control name="lastName" type="text" required onChange={handleChange} />
                                <Form.Text className="text-muted">Last name</Form.Text>
                                <Form.Control.Feedback type="invalid">
                                    Please enter your last name.
                                </Form.Control.Feedback>
                            </Form.Group>
                        </Col>
                    </Form.Row>

                    <Form.Group controlId="contactEmail">
                        <Form.Label>Email *</Form.Label>
                        <Form.Control name="email" type="email" required onChange={handleChange} />
                        <Form.Control.Feedback type="invalid">
                            Please enter a valid email address.
                        </Form.Control.Feedback>
                    </Form.Group>

                    <Form.Group controlId="contactSubject">
                        <Form.Label>Subject *</Form.Label>
                        <Form.Control name="subject" type="text" required onChange={handleChange} />
                        <Form.Control.Feedback type="invalid">Please enter a subject.</Form.Control.Feedback>
                    </Form.Group>

                    <Form.Group controlId="contactMessage">
                        <Form.Label>Message *</Form.Label>
                        <textarea
                            name="message"
                            className="form-control"
                            id="contactMessage"
                            rows="5"
                            required
                            onChange={handleChange}
                        />
                        <Form.Control.Feedback type="invalid">Please enter your message.</Form.Control.Feedback>
                    </Form.Group>

                    <Form.Group controlId="contactRecaptcha">
                        <Recaptcha
                            id="contactRecaptcha"
                            ref={recaptchaRef}
                            sitekey={RECAPTCHA_KEY}
                            onChange={handleRecaptcha}
                            className={Styles.contactFormReCaptcha}
                        />
                        {true === recaptchaError && (
                            <div className="invalid-feedback" style={{ display: 'block' }}>
                                Please answer the captcha.
                            </div>
                        )}
                    </Form.Group>

                    <Button variant="primary" type="submit" className={Styles.contactFormSubmitButton}>
                        Submit
                    </Button>
                </Form>
            </Fade>
            <Fade
                collapse
                when={success}
                duration={SUCCESS_MESSAGE_FADE_IN_DURATION}
                delay={CONTACT_FORM_FADE_UP_DURATION}
            >
                <Alert variant="success" className={Styles.contactFormSuccess}>
                    <p>{`Thank you for your message. We're getting in touch with you soon.`}</p>
                </Alert>
            </Fade>
        </div>
    );
};
