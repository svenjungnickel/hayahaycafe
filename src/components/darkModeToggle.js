import React from 'react';
import useDarkMode from 'use-dark-mode';
import ReactSwitch from 'react-switch';

const Switch = props => <ReactSwitch {...props} />;

Switch.defaultProps = {
    checkedIcon: false,
    uncheckedIcon: false,
    height: 24,
    width: 48,
    handleDiameter: 24,
    offColor: `#000`,
    onColor: `#000`,
    boxShadow: `inset 0 0 0 1px #000`,
};

const uncheckedIcon = (
    <div
        style={{
            display: 'flex',
            justifyContent: 'center',
            alignItems: 'center',
            height: '100%',
            fontSize: 15,
            color: 'yellow',
            paddingLeft: 2,
        }}
    >
        ☀
    </div>
);

const checkedIcon = (
    <div
        style={{
            display: 'flex',
            justifyContent: 'center',
            alignItems: 'center',
            height: '100%',
            fontSize: 15,
            color: 'yellow',
            paddingRight: 2,
        }}
    >
        ☾
    </div>
);

const DarkModeToggle = props => {
    const darkMode = useDarkMode(false),
        darkModeValue = null !== darkMode.value ? darkMode.value : false;

    return (
        <Switch
            {...props}
            title="dark mode switch"
            aria-checked={darkModeValue}
            checked={darkModeValue}
            onChange={darkMode.toggle}
            uncheckedIcon={uncheckedIcon}
            checkedIcon={checkedIcon}
        />
    );
};

export default DarkModeToggle;
