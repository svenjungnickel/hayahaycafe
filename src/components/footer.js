import React from 'react';
import { Container, Nav } from 'react-bootstrap';
import SocialMediaIcons from './socialMediaIcons';
import FooterStyles from '../styles/components/footer.module.scss';

export default ({ siteMetadata }) => (
    <footer className={FooterStyles.footer}>
        <div className={FooterStyles.footer__pre}>
            <SocialMediaIcons items={siteMetadata.socialMedia} />
        </div>
        <div className={FooterStyles.footer__main}>
            <Container className="text-center">
                <Nav className="justify-content-center">
                    <Nav.Item>
                        <Nav.Link href="/contact">Contact</Nav.Link>
                    </Nav.Item>
                    <Nav.Item>
                        <Nav.Link href="/legal">Legal</Nav.Link>
                    </Nav.Item>
                    <Nav.Item>
                        <Nav.Link href="/data-privacy">Data Privacy</Nav.Link>
                    </Nav.Item>
                </Nav>

                <div className={FooterStyles.footer__address}>
                    <p>{siteMetadata.address}</p>
                </div>
            </Container>
        </div>
    </footer>
);
