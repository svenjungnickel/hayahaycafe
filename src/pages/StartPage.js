import React from 'react';
import PropTypes from 'prop-types';
import { graphql } from 'gatsby';
import SEO from '../components/seo';
import Layout from '../components/layout';
import { HTMLContent } from '../components/content';
import StartPageTemplate from '../templates/StartPageTemplate';

const StartPage = ({ data: { page } }) => (
    <Layout currentPage="/">
        <SEO
            title={page.frontmatter.title}
            keywords={(page.frontmatter.meta && page.frontmatter.meta.keywords) || []}
        />
        {/*<SEO meta={page.frontmatter.meta || false} />*/}

        <StartPageTemplate {...page.frontmatter} content={page.html} contentComponent={HTMLContent} />
    </Layout>
);

StartPage.propTypes = {
    data: PropTypes.shape({
        markdownRemark: PropTypes.object,
    }),
};

export default StartPage;

export const pageQuery = graphql`
    ## Query for HomePage data
    ## Use GraphiQL interface (http://localhost:8000/___graphql)
    ## $id is processed via gatsby-node.js
    ## query name must be unique to this file
    query IndexPage($id: String!) {
        page: markdownRemark(id: { eq: $id }) {
            html
            frontmatter {
                title
                subtitle
                location
                openingHours
                headerImage {
                    childImageSharp {
                        fluid {
                            ...GatsbyImageSharpFluid
                        }
                    }
                    publicURL
                }
            }
        }
    }
`;
