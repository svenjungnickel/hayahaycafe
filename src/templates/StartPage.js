import React from 'react';
import PropTypes from 'prop-types';
import { graphql } from 'gatsby';
import { Col, Container, Row } from 'react-bootstrap';
import SEO from '../components/seo';
import Layout from '../components/layout';
import Image from '../components/image';
import Gallery from '../components/gallery';
import Content, { HTMLContent } from '../components/content';
import IndexStyles from '../styles/pages/index.module.scss';
import SectionStyles from '../styles/components/section.module.scss';

export const StartPageTemplate = ({
    title,
    subtitle,
    headerImage,
    content,
    contentComponent,
    location,
    openingHours,
    gallery,
}) => {
    const PostContent = contentComponent || Content;

    return (
        <>
            <header className={IndexStyles.indexHeader} id="home">
                <Image src={headerImage} alt={title} className={IndexStyles.indexHeader__image} />

                <div className={IndexStyles.indexHeader__content}>
                    <Container>
                        <div className={IndexStyles.indexHeader__contentInner}>
                            <h1 className="display-1">{title}</h1>
                            <p className="lead">{subtitle}</p>
                        </div>
                    </Container>
                </div>
            </header>

            <section className={SectionStyles.section} id="cafe">
                <Container>
                    <Row>
                        <Col xs={12} sm={9} lg={8}>
                            <PostContent content={content} />
                        </Col>
                        <Col xs={12} sm={1} lg={1}>
                            &nbsp;
                        </Col>
                        <Col xs={12} sm={2} lg={3}>
                            <h4>Location</h4>
                            <PostContent content={location} />
                            <p>&nbsp;</p>
                            <h4>Hours</h4>
                            <PostContent content={openingHours} />
                        </Col>
                    </Row>
                </Container>
            </section>

            <Gallery images={gallery} />
        </>
    );
};

StartPageTemplate.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string.isRequired,
    headerImage: PropTypes.oneOfType([PropTypes.object, PropTypes.string]).isRequired,
    content: PropTypes.node.isRequired,
    contentComponent: PropTypes.func,
    location: PropTypes.node.isRequired,
    openingHours: PropTypes.node.isRequired,
    gallery: PropTypes.array,
};

const StartPage = ({ data: { page } }) => (
    <Layout currentPage="/">
        <SEO
            title={page.frontmatter.title}
            keywords={(page.frontmatter.meta && page.frontmatter.meta.keywords) || []}
        />
        {/*<SEO meta={page.frontmatter.meta || false} />*/}

        <StartPageTemplate {...page.frontmatter} content={page.html} contentComponent={HTMLContent} />
    </Layout>
);

StartPage.propTypes = {
    data: PropTypes.shape({
        markdownRemark: PropTypes.object,
    }),
};

export default StartPage;

export const pageQuery = graphql`
    ## Query for HomePage data
    ## Use GraphiQL interface (http://localhost:8000/___graphql)
    ## $id is processed via gatsby-node.js
    ## query name must be unique to this file
    query IndexPage($id: String!) {
        page: markdownRemark(id: { eq: $id }) {
            html
            frontmatter {
                title
                subtitle
                location
                openingHours
                headerImage {
                    childImageSharp {
                        fluid {
                            ...GatsbyImageSharpFluid
                        }
                    }
                    publicURL
                }
                gallery {
                    src {
                        childImageSharp {
                            fluid(maxWidth: 1600, maxHeight: 800) {
                                ...GatsbyImageSharpFluid
                            }
                        }
                        publicURL
                    }
                    alt
                    title
                }
            }
        }
    }
`;
