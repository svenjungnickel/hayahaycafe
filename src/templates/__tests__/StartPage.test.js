import React from 'react';
import renderer from 'react-test-renderer';
import StartPage, { StartPageTemplate } from '../StartPage';
import { HTMLContent } from '../../components/Content';
import StartPageData from '../../__fixtures__/StartPageData';

jest.mock('../../components/Layout');
jest.mock('../../components/Image');
jest.mock('../../components/Gallery');
jest.mock('../../components/Header');
jest.mock('../../components/Location/Location');
jest.mock('../../components/Separator');
jest.mock('../../components/OpeningHours');
jest.mock('../../components/What3WordsAddress');

describe('StartPageTemplate', () => {
    it('renders without title throws prop type error', () => {
        const renderStartPageTemplate = () => {
            const subtitle = 'subtitle';
            const headerImage = 'headerImage';
            const content = 'content';
            const gallery = [{}];

            const component = (
                <StartPageTemplate subtitle={subtitle} headerImage={headerImage} content={content} gallery={gallery} />
            );
            renderer.create(component);
        };

        expect(renderStartPageTemplate).toThrowError('Warning: Failed prop type');
    });

    it('renders without subtitle throws prop type error', () => {
        const renderStartPageTemplate = () => {
            const title = 'title';
            const headerImage = 'headerImage';
            const content = 'content';
            const gallery = [{}];

            const component = (
                <StartPageTemplate title={title} headerImage={headerImage} content={content} gallery={gallery} />
            );
            renderer.create(component);
        };

        expect(renderStartPageTemplate).toThrowError('Warning: Failed prop type');
    });

    it('renders without headerImage throws prop type error', () => {
        const renderStartPageTemplate = () => {
            const title = 'title';
            const subtitle = 'subtitle';
            const content = 'content';
            const gallery = [{}];

            const component = (
                <StartPageTemplate title={title} subtitle={subtitle} content={content} gallery={gallery} />
            );
            renderer.create(component);
        };

        expect(renderStartPageTemplate).toThrowError('Warning: Failed prop type');
    });

    it('renders without content throws prop type error', () => {
        const renderStartPageTemplate = () => {
            const title = 'title';
            const subtitle = 'subtitle';
            const headerImage = 'headerImage';
            const gallery = [{}];

            const component = (
                <StartPageTemplate title={title} subtitle={subtitle} headerImage={headerImage} gallery={gallery} />
            );
            renderer.create(component);
        };

        expect(renderStartPageTemplate).toThrowError('Warning: Failed prop type');
    });

    it('renders without gallery throws prop type error', () => {
        const renderStartPageTemplate = () => {
            const title = 'title';
            const subtitle = 'subtitle';
            const headerImage = 'headerImage';
            const content = 'content';

            const component = (
                <StartPageTemplate title={title} subtitle={subtitle} headerImage={headerImage} content={content} />
            );
            renderer.create(component);
        };

        expect(renderStartPageTemplate).toThrowError('Warning: Failed prop type');
    });

    it('renders with invalid title throws prop type error', () => {
        const renderStartPageTemplate = () => {
            const title = 123;
            const subtitle = 'subtitle';
            const headerImage = 'headerImage';
            const content = 'content';
            const gallery = [{}];

            const component = (
                <StartPageTemplate
                    title={title}
                    subtitle={subtitle}
                    headerImage={headerImage}
                    content={content}
                    gallery={gallery}
                />
            );
            renderer.create(component);
        };

        expect(renderStartPageTemplate).toThrowError('Warning: Failed prop type');
    });

    it('renders with invalid subtitle throws prop type error', () => {
        const renderStartPageTemplate = () => {
            const title = 'title';
            const subtitle = 123;
            const headerImage = 'headerImage';
            const content = 'content';
            const gallery = [{}];

            const component = (
                <StartPageTemplate
                    title={title}
                    subtitle={subtitle}
                    headerImage={headerImage}
                    content={content}
                    gallery={gallery}
                />
            );
            renderer.create(component);
        };

        expect(renderStartPageTemplate).toThrowError('Warning: Failed prop type');
    });

    it('renders with invalid headerImage throws prop type error', () => {
        const renderStartPageTemplate = () => {
            const title = 'title';
            const subtitle = 'subtitle';
            const headerImage = 123;
            const content = 'content';
            const gallery = [{}];

            const component = (
                <StartPageTemplate
                    title={title}
                    subtitle={subtitle}
                    headerImage={headerImage}
                    content={content}
                    gallery={gallery}
                />
            );
            renderer.create(component);
        };

        expect(renderStartPageTemplate).toThrowError('Warning: Failed prop type');
    });

    it('renders with invalid content throws prop type error', () => {
        const renderStartPageTemplate = () => {
            const title = 'title';
            const subtitle = 'subtitle';
            const headerImage = 'headerImage';
            const content = {};
            const gallery = [{}];

            const component = (
                <StartPageTemplate
                    title={title}
                    subtitle={subtitle}
                    headerImage={headerImage}
                    content={content}
                    gallery={gallery}
                />
            );
            renderer.create(component);
        };

        expect(renderStartPageTemplate).toThrowError('Warning: Failed prop type');
    });

    it('renders with invalid contentComponent throws prop type error', () => {
        const renderStartPageTemplate = () => {
            const title = 'title';
            const subtitle = 'subtitle';
            const headerImage = 'headerImage';
            const content = 'content';
            const contentComponent = false;
            const gallery = [{}];

            const component = (
                <StartPageTemplate
                    title={title}
                    subtitle={subtitle}
                    headerImage={headerImage}
                    content={content}
                    contentComponent={contentComponent}
                    gallery={gallery}
                />
            );
            renderer.create(component);
        };

        expect(renderStartPageTemplate).toThrowError('Warning: Failed prop type');
    });

    it('renders with invalid gallery throws prop type error', () => {
        const renderStartPageTemplate = () => {
            const title = 'title';
            const subtitle = 'subtitle';
            const headerImage = 'headerImage';
            const content = 'content';
            const gallery = false;

            const component = (
                <StartPageTemplate
                    title={title}
                    subtitle={subtitle}
                    headerImage={headerImage}
                    content={content}
                    gallery={gallery}
                />
            );
            renderer.create(component);
        };

        expect(renderStartPageTemplate).toThrowError('Warning: Failed prop type');
    });

    it('renders content correctly', () => {
        const title = 'title';
        const subtitle = 'subtitle';
        const headerImage = 'headerImage';
        const content = 'content';
        const gallery = [{}];

        const component = (
            <StartPageTemplate
                title={title}
                subtitle={subtitle}
                headerImage={headerImage}
                content={content}
                gallery={gallery}
            />
        );
        const tree = renderer.create(component).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('renders html content correctly', () => {
        const title = 'title';
        const subtitle = 'subtitle';
        const headerImage = 'headerImage';
        const content = <p>HTML content</p>;
        const gallery = [{}];

        const component = (
            <StartPageTemplate
                title={title}
                subtitle={subtitle}
                headerImage={headerImage}
                content={content}
                contentComponent={HTMLContent}
                gallery={gallery}
            />
        );
        const tree = renderer.create(component).toJSON();

        expect(tree).toMatchSnapshot();
    });
});

describe('StartPage', () => {
    it('renders without data throws prop type error', () => {
        const renderStartPage = () => {
            const component = <StartPage />;
            renderer.create(component);
        };

        expect(renderStartPage).toThrowError('Warning: Failed prop type');
    });

    it('renders with invalid data throws prop type error', () => {
        const renderStartPage = () => {
            const data = {};

            const component = <StartPage data={data} />;
            renderer.create(component);
        };

        expect(renderStartPage).toThrowError('Warning: Failed prop type');
    });

    it('renders correctly', () => {
        const data = {
            page: {
                html: <p>HTML</p>,
                frontmatter: StartPageData,
            },
        };

        const component = <StartPage data={data} />;
        const tree = renderer.create(component).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
