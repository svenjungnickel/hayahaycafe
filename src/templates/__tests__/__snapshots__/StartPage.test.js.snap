// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StartPage renders correctly 1`] = `
<main>
  <header
    className="startHeader"
  >
    <img
      alt="alt"
      src="src"
    />
    <div
      className="startHeader__content"
    >
      <div
        className="container"
      >
        <div
          className="startHeader__contentInner"
        >
          <h1
            className="display-1"
          >
            title
          </h1>
          <p
            className="lead"
          >
            subtitle
          </p>
        </div>
      </div>
    </div>
  </header>
  <section
    className="section"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-12"
        >
          <div
            dangerouslySetInnerHTML={
              Object {
                "__html": <p>
                  HTML
                </p>,
              }
            }
          />
          <hr />
        </div>
        <div
          className="col-sm-4 col-12"
          data-cy="startPageLocation"
        >
          <p>
            Location
          </p>
        </div>
        <div
          className="col-sm-4 col-12"
          data-cy="startPageWhat3Words"
        >
          <hr />
          <a
            href="what3WordsAddress"
          >
            <img
              alt="what3WordsIcon"
              src="src"
            />
          </a>
        </div>
        <div
          className="col-sm-4 col-12"
          data-cy="startPageOpeningHours"
        >
          <hr />
          <p>
            Opening Hours
          </p>
        </div>
        <div
          className="col-12"
        >
          <hr />
        </div>
        <div
          className="container"
        >
          <h2
            className="display-5 text-center"
          >
            Gallery
          </h2>
          <p>
            Gallery
          </p>
        </div>
      </div>
    </div>
  </section>
</main>
`;

exports[`StartPageTemplate renders content correctly 1`] = `
Array [
  <header
    className="startHeader"
  >
    <img
      alt="alt"
      src="src"
    />
    <div
      className="startHeader__content"
    >
      <div
        className="container"
      >
        <div
          className="startHeader__contentInner"
        >
          <h1
            className="display-1"
          >
            title
          </h1>
          <p
            className="lead"
          >
            subtitle
          </p>
        </div>
      </div>
    </div>
  </header>,
  <section
    className="section"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-12"
        >
          <div>
            content
          </div>
          <hr />
        </div>
        <div
          className="col-sm-4 col-12"
          data-cy="startPageLocation"
        >
          <p>
            Location
          </p>
        </div>
        <div
          className="col-sm-4 col-12"
          data-cy="startPageWhat3Words"
        >
          <hr />
          <a
            href="what3WordsAddress"
          >
            <img
              alt="what3WordsIcon"
              src="src"
            />
          </a>
        </div>
        <div
          className="col-sm-4 col-12"
          data-cy="startPageOpeningHours"
        >
          <hr />
          <p>
            Opening Hours
          </p>
        </div>
        <div
          className="col-12"
        >
          <hr />
        </div>
        <div
          className="container"
        >
          <h2
            className="display-5 text-center"
          >
            Gallery
          </h2>
          <p>
            Gallery
          </p>
        </div>
      </div>
    </div>
  </section>,
]
`;

exports[`StartPageTemplate renders html content correctly 1`] = `
Array [
  <header
    className="startHeader"
  >
    <img
      alt="alt"
      src="src"
    />
    <div
      className="startHeader__content"
    >
      <div
        className="container"
      >
        <div
          className="startHeader__contentInner"
        >
          <h1
            className="display-1"
          >
            title
          </h1>
          <p
            className="lead"
          >
            subtitle
          </p>
        </div>
      </div>
    </div>
  </header>,
  <section
    className="section"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-12"
        >
          <div
            dangerouslySetInnerHTML={
              Object {
                "__html": <p>
                  HTML content
                </p>,
              }
            }
          />
          <hr />
        </div>
        <div
          className="col-sm-4 col-12"
          data-cy="startPageLocation"
        >
          <p>
            Location
          </p>
        </div>
        <div
          className="col-sm-4 col-12"
          data-cy="startPageWhat3Words"
        >
          <hr />
          <a
            href="what3WordsAddress"
          >
            <img
              alt="what3WordsIcon"
              src="src"
            />
          </a>
        </div>
        <div
          className="col-sm-4 col-12"
          data-cy="startPageOpeningHours"
        >
          <hr />
          <p>
            Opening Hours
          </p>
        </div>
        <div
          className="col-12"
        >
          <hr />
        </div>
        <div
          className="container"
        >
          <h2
            className="display-5 text-center"
          >
            Gallery
          </h2>
          <p>
            Gallery
          </p>
        </div>
      </div>
    </div>
  </section>,
]
`;
